tomplot/local_area_plot.py

```diff
@@ -0,0 +1,35 @@
+import pandas as pd
+
+
+def area_restriction(field_data, coords_X, coords_Y, Y_lim=None, X_lim=None):
+    """
+    a function that takes a data dictionary and will restrict it for a given value
+    For the coordinate fields you must pass in a 1d object so either coords[:,level]
+    for slicing on z or coords.flatten if slicing along lat / lon
+
+    args:
+    field_data: The Plotting field you wish to return
+    coords_X: The coordinates you wish to plot along the X axis.
+    coords_Y: The field you wish to plot along the Y axis.
+    X_lim: tuple containing the (lower, upper) limits for the X axis
```

Comment:
To make this more general, we could combine `X_lim` and `Y_lim` to be a single `coords_lims` dictionary

Reply:
I think this is a very good idea and will implement this. 
The one thing that i am not too sure about is how we would handle having a plot in which you wanted different areas highlighted? For example you may wish to have a 1x2 plot where the first plot is a northern hemisphere plot and the second is a southern hemisphere plot.
Would we need another key? Or can you iterate through dictionaries in a list? 
Ill implement it in the case that its the same area for all plots until you get back to me about this